google/cloud/aiplatform/training_jobs.py

```diff
@@ -1753,3 +1754,329 @@ def _model_upload_fail_string(self) -> str:
+                cost will be attempted to be close to the budget, though may end up
+                being (even) noticeably smaller - at the backend's discretion. This
+                especially may happen when further model training ceases to provide
+                any improvements.
+                If the budget is set to a value known to be insufficient to train a
+                Model for the given training set, the training won't be attempted and
+                will error.
+                The minimum value is 1000 and the maximum is 72000.
+            model_display_name (str):
+                Optional. The display name of the managed AI Platform Model. The name
+                can be up to 128 characters long and can be consist of any UTF-8
+                characters. If a `base_model` was provided, the display_name in the
+                base_model will be overritten with this value. If not provided upon
+                creation, the job's display_name is used.
+            disable_early_stopping (bool):
+                Required. If true, the entire budget is used. This disables the early stopping
+                feature. By default, the early stopping feature is enabled, which means
+                that training might stop before the entire training budget has been
+                used, if further training does no longer brings significant improvement
+                to the model.
+            sync (bool):
+                Whether to execute this method synchronously. If False, this method
+                will be executed in concurrent Future and any downstream object will
+                be immediately returned and synced when the Future has completed.
+
+        Returns:
+            model: The trained AI Platform Model resource or None if training did not
+                produce an AI Platform Model.
+        """
+
+        # Retrieve the objective-specific training task schema based on prediction_type
+        training_task_definition = getattr(
+            schema.training_job.definition, f"automl_image_{self._prediction_type}_task"
+        )
+
+        training_task_inputs_dict = {
+            # required inputs
+            "modelType": self._model_type,
+            "budgetMilliNodeHours": budget_milli_node_hours,
+            "modelLabel": self._multi_label,
+            # optional inputs
+            "disableEarlyStopping": disable_early_stopping,
+        }
+
+        if model_display_name is None:
+            model_display_name = self._display_name
+
+        model_tbt = None  # gca Model to be trained
+
+        if not self._base_model:
+            model_tbt = gca_model.Model(display_name=model_display_name)
+        else:
+            # Use provided base_model to pass to model_to_upload causing the
+            # description and tags from base_model to be passed onto the new model
+            model_tbt = self._base_model._gca_resource
```

Comment:
This seems like it should be copied or this will mutate the display name in the Model instance's gca_resource.